test/interactions/interactionTests.ts

```diff
@@ -0,0 +1,110 @@
+///<reference path="../testReference.ts" />
+
+var assert = chai.assert;
+
+describe("Interactions", () => {
+  describe("Interactions", () => {
+    var SVG_WIDTH = 400;
+    var SVG_HEIGHT = 400;
+
+    it("attaching/detaching a component modifies the state of the interaction", () => {
+      var svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var component = new Plottable.Component();
+      var interaction = new Plottable.Interaction();
+      component.renderTo(svg);
+
+      interaction.attachTo(component);
+      assert.strictEqual((<any>interaction)._componentAttachedTo, component,
+        "the _componentAttachedTo field should contain the component the interaction is attached to");
+
+      interaction.detachFrom(component);
+      assert.isNull((<any>interaction)._componentAttachedTo,
+        "the _componentAttachedTo field should be blanked upon detaching");
+
+      svg.remove();
+    });
+
+    it("can attach/detach interaction to/from component", () => {
+      var svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      var component = new Plottable.Component();
+      component.renderTo(svg);
+
+      var clickInteraction = new Plottable.Interactions.Click();
+
+      var callbackCalled = false;
+      var callback = () => callbackCalled = true;
+      clickInteraction.onClick(callback);
+
+      clickInteraction.attachTo(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isTrue(callbackCalled, "callback called on clicking Component (mouse)");
+
+      callbackCalled = false;
+      clickInteraction.detachFrom(component);
+
+      TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      assert.isFalse(callbackCalled, "callback was removed from component and should not be called");
+
+      svg.remove();
+    });
+
+    it("interactions are reusable", () => {
```

Comment:
if we _should_ revisit the tests, then why not bring up the discussion again then and leave the tests like this for now.